### DISCUSSION TITLE
Environment variable to control the default output style

### DISCUSSION THREAD

**Discussion by jonathanmorley** [nrwl/nx#27490] [Feature Requests]
I prefer using `--output-style=stream` locally, so that i can see and track the output of targets (and dependencies).

I would like to be able to make this a system-wide preference, rather than needing to add it when invoking commands.

I would imagine something like `NX_DEFAULT_OUTPUT_STYLE` in terms of naming.

**Comment by flo-sch** [comment]
This would also be handy in CI environments, where it could be controlled from a single variable instead of having to specify `--output-style=static` to each command manually

**Comment by bustinjailey** [comment]
I would also find this very convenient locally and for CI pipelines

**Comment by mrkimbo** [comment]
Also adding this as a configuration prop in `nx.json` would be good (`defaultOutputStyle` or `outputStyle`)

**Comment by cfairbanks** [comment]
This would be extremely useful for CI pipelines - the new nx 21 client UI frequently crashes my CI builds and I have to add ` --output-style=dynamic-legacy` manually to so many individual commands.

**Comment by gabrielbryk** [comment]
bump this

**Comment by avens19** [comment]
https://github.com/nrwl/nx/pull/33493
